### PR TITLE
interactive: add function that counts number of operations in a module

### PR DESCRIPTION
**Pull request by dshaaban01**
Add function that takes a ModuleOp and returns a dictionary containing the number of occurences of each Operation in the ModuleOp. First step to provide users with feedback on whether the passes/ optimisations they have chosen are "good" or "bad".

### CODE REVIEW

**Comment by dshaaban01** [comment]
@webmiche What do you think? We briefly discussed smth like this in our last meeting. Thoughts?

File: xdsl/interactive/pass_metrics.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    # Apply .name to each element in the list using a list comprehension
+    names_list = [op.name for op in all_operations]
+
+    # Count occurrences of each unique Operation
+    operation_counter = Counter(names_list)
+
+    res_operations_count: dict[str, int] = dict(operation_counter.items())
+
+    return res_operations_count
```
Comment (superlopuh) [inline xdsl/interactive/pass_metrics.py:25]:
```suggestion
    return Counter(op.name for op in module.walk())
```

Comment (superlopuh) [reply xdsl/interactive/pass_metrics.py:25]: Might need to change the return type to `Mapping[str, int]`

Comment (superlopuh) [reply xdsl/interactive/pass_metrics.py:25]: But probably not since `Counter` subclasses `dict`

Comment (dshaaban01) [reply xdsl/interactive/pass_metrics.py:25]: Wow. You really made 13 lines of code redundant. That's optimization if I ever saw it.

Comment (superlopuh) [reply xdsl/interactive/pass_metrics.py]: The power of Python :)

**Review by superlopuh — APPROVED** [review]
Looks great!

**Review by webmiche — COMMENTED** [review]
Nice, LGTM.

**Pull request merged**